### PR TITLE
Shortening events with add variant (Wanderer Defense + Mk2 Republic)

### PR DESCRIPTION
**Pull request by tehhowch**
One fleet got missed (Wanderer Defense). I also incorporated the #2577 functionality to change the mk2 republic event from redefinition to dynamic update. Using the dynamic method adds back a few lines, but will prevent squashing any content added to these fleets by plugins, and also help introduce this syntax & use to content-creators.

### CODE REVIEW

File: data/events.txt [hunk 1]
```diff
@@ -1078,65 +1078,105 @@ event "fw safe passage ends"
 
 event "navy using mark ii ships"
 	fleet "Small Republic"
-		variant 4
+		remove variant 6
```
Comment (endless-sky) [inline data/events.txt]: For removing variants, you don't need to specify the weight.

Comment (tehhowch) [inline data/events.txt]: Yes, I know. I had added them for clarity - without looking back to the original fleet definition, one knows the original variant's weight.

**Comment by tehhowch** [comment]
Yes, using both `add variant` and `remove variant` to maintain a dynamic definition of the fleet will probably result in longer definitions than using just `variant` (particularly when most of the variants are being altered). The benefit here is that all variants not identified are kept.For the Small/Large Republic fleets in the base game there are no such variants, but there are several mods that add to these definitions, and these changes mean the mk2 fleet update will not remove those additions.

**Comment by endless-sky** [comment]
I think it'd make a lot more sense to just give the newly added variants higher weights, rather than removing and then re-adding old variants just to reduce their weights. For example, the Small Republic modification could just do this:

```
        add variant 12
            "Rainmaker (Mark II)" 2
        add variant 12
            "Gunboat (Mark II)"
            "Rainmaker (Mark II)"
        add variant 9
            "Frigate (Mark II)"
        add variant 6
            "Gunboat (Mark II)" 2
        add variant 3
            "Rainmaker (Mark II)" 3
        add variant 12
            "Gunboat (Mark II)"
```

That would be identical to the existing fleet definition (with all weights multiplied by 3) except that the weight for the old Gunboat + old Rainmaker combo would increase from 3 to 5.

Because the weights are probabilities anyway (just trying to roughly sketch out what variants should be most frequent) it's not important for a change to the definitions to keep them *exactly* the same.

**Comment by tehhowch** [comment]
@endless-sky done with cd0c784

**Pull request merged**